modules/storages/app/common/storages/peripherals/storage_interaction/authentication_strategies/oauth_client_credentials.rb

```diff
@@ -59,12 +59,16 @@ def call(storage:, http_options: {})
 
             http = http_result.result
 
-            if access_token.nil?
+            operation_result = yield http
+
+            if access_token.nil? && operation_result.success?
               token = http.instance_variable_get(:@options).oauth_session.access_token
               Rails.cache.write("storage.#{storage.id}.httpx_access_token", token, expires_in: 50.minutes)
+            elsif operation_result.failure? && operation_result.result == :forbidden
+              Rails.cache.delete("storage.#{storage.id}.httpx_access_token")
```

Comment:
Let's extract cache key construction to a method or save it to a local variable to have it defined just in one place.
It will help to avoid stupid error when accidentally cache key got changed only in one place, but not in the other.

Reply:
Good point.